LibertyDSNP/spec
# Feat/convert mdbook/144

Problem
=======
Our static site generator (docz) isn't really meeting our needs currently. 
#144

Solution
========
Using mdBook as our static site generator to take advantage of improved functionality: 
- Nested menus in the sidebar
- Multiple themes
- Better configuration opportunities
- Multiple third-party plugins

with @wilwade 

Change summary:
---------------
This is Pull request 1 of 3. See comment for next steps
- [x] Updated dependencies in package.json
- [x] Removed docz and generated spec with mdBook
- [x] Configured styles to have a cleaner look
- [x] Add custom var for sidebar menu folding and collapsing nested menus
- [x] Added third-party plugin [mdBook-linkcheck](https://github.com/Michael-F-Bryan/mdbook-linkcheck) for checking broken links 
- [x] Added github repo URL to the menu bar for more transparency 
- [x] Replaced absolute URLs with relative ones 
- [x] Used provided handlebars to add DSNP logo, and external link to the developer portal 

Steps to Verify:
----------------
From root dir:
1. npm install clean
1. npm run build
2. Build process will run a link-check to check for broken links
1. npm run lint
1. npm run serve
1. The spec will automatically open in your web browser
1. Click around for internal and external links

Screenshots (optional):
-----------------------
<img width="1346" alt="image" src="https://user-images.githubusercontent.com/40524916/160213083-832f34ac-41a6-4fed-8e9b-85cb1fea023a.png">
